hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/WritableUpgradeStore.java

```diff
@@ -33,11 +34,11 @@
  * <p>This class is not exported from the module. It is an internal implementation detail. This
  * class is not complete, it will be extended with other methods like remove, update etc.,
  */
-public class WritableUpgradeStore extends ReadableUpgradeStoreImpl implements WritableQueueState<File> {
+public class WritableUpgradeStore extends ReadableUpgradeStoreImpl {
     /** The underlying data storage class that holds the file data. */
     private final WritableQueueState<Bytes> writableUpgradeState;
 
-    private final WritableSingletonState<File> writableUpgradeFileState;
+    private final WritableKVState<FileID, File> writableUpgradeFileState;
```

Comment:
Do we need this store now ? Can we not use `WritableFileStore` , since it almost seems the same ? Am I missing something

Reply:
Upgrade file has to use the WritableQueueState for the file content.

Reply:
It is not implementing `WritableQueueState` anymore. So how does it store the content in `WritableQueueState` ? Is it added somewhere else?

Reply:
It absolutely does still read from the queue state.  The `File` is in KV state, the *content* (the bytes of the file) are still filled in from the queue state.  If it did not do that it would level the node.
Look at lines 39, 58, 64, etc...  The readable mirror is similar.
There are two states in each class (WritableUpgradeState/WritableUpgradeFileState and ReadableUpgradeState/ReadableUpgradeFileState).